tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -117,8 +117,13 @@ def displayed_app(self):
                        origin=result.get('origin'))
 
     def switch_to_displayed_app(self):
+        from gaiatest.apps.lockscreen.app import LockScreen
+
         self.marionette.switch_to_default_content()
-        self.marionette.switch_to_frame(self.displayed_app.frame)
+        displayed_app = self.displayed_app
+        if displayed_app.origin != LockScreen.origin:
+            # The lockscreen app is not in an iFrame. It's a div in the System app.
+            self.marionette.switch_to_frame(self.displayed_app.frame)
```

Comment:
Actually, I wonder if you really need to special case the lockscreen case here. 
What does self.displayed_app.frame say in the case when lockscreen visible? Perhaps you can just call self.marionette.switch_to_frame(self.displayed_app.frame) directly or if self.displayed_app.frame returns undefined or something, don't do anything?